### PR TITLE
CI: improve CI Tests files blacklist

### PR DESCRIPTION
**Pull request by khancyr**
Alternative to https://github.com/ArduPilot/ardupilot/pull/22885

Add more files to blacklist on test to speed up our PR testing.

### CODE REVIEW

**Comment by peterbarker** [comment]
Random thought: could we test these blacklists by actually *removing* the files before compiling?

It won't catch everything (files which modify the build process through their existence / are auto-detected, but it would catch gross errors.

Wouldn't take long to do it on every build, either.

I'm not suggesting this be done on this PR - but it would make for a great test!

File: .github/workflows/test_sitl_copter.yml [hunk 1]
```diff
@@ -2,20 +2,135 @@ name: test copter
...
+      - 'Tools/FilterTestTool/**'
+      - 'Tools/Frame_params/**'
+      - 'Tools/GIT_Test/**'
+      - 'Tools/Hello/**'
+      - 'Tools/IO_Firmware/**'
+      - 'Tools/Linux_HAL_Essentials/**'
+      - 'Tools/LogAnalyzer/**'
+      - 'Tools/Pozyx/**'
+      - 'Tools/PrintVersion.py'
+      - 'Tools/Replay/**'
```
Comment (peterbarker) [inline .github/workflows/test_sitl_copter.yml:30]: This doesn't look right; we test Replay as part of our Copter SITL testing, don't we?

**Review by peterbarker — CHANGES_REQUESTED** [review]
Having found the Replay exclusion here, I think to have confidence here we do need to delete these paths before testing.

File: .github/workflows/test_sitl_copter.yml (hunk 1 above)
Comment (khancyr) [reply .github/workflows/test_sitl_copter.yml:30]: Rigth, I wasn't aware, we got a Replay workflow  https://github.com/ArduPilot/ardupilot/blob/master/.github/workflows/test_replay.yml so I didn't think that replay was hidden into copter test. We should probably correct this into antoher PR

**Comment by khancyr** [comment]
Tested by removing blacklist files as much as possible, in some case we got inclusion even if not used so we need to keeps the files, but that don't collide with the test.

So now test are working correctly.

**Comment by khancyr** [comment]
Reordering and moving some of our files would allow better blacklisting. As the drawback of blacklist is that we cannot have exception, so if we blacklist a directory that is all the files in it ... 

Another things we could test later is to use a whitelist with everything include (**/**) and then use blacklist to remove things from it. Not sure if it works thus. That would allow use to have some exception on whitelist. 
And example is chibios test that need test_build_options.py from Tools/autotest and pysim/util.py, then we cannot blacklist Tools/autotest directory, whereas we need only 2 files from it. So a change on locations.txt will trigger full chibios test....

Improvements for another day

**Comment by peterbarker** [comment]
Please rebase and push this one

**Review by tridge — APPROVED** [review]

**Pull request merged**